lib/mix_unused/analyzers/unreachable.ex

```diff
@@ -27,6 +27,7 @@ defmodule MixUnused.Analyzers.Unreachable do
         mfa not in usages,
         mfa not in reachables,
         mfa not in called_at_compile_time,
+        config.report_transitively_used or Graph.in_degree(graph, mfa) == 0,
```

Comment:
Here Elixir does not know that `config` is the `Config` struct.
You can add the following above:
```
config = %Config{} = Config.cast(config)
```

Then Elixir should report a compile-time error if the field does not exist.

You can also consider to pattern match:
```
config = %Config{report_transitively_used: report_transitively_used} = Config.cast(config)
```

As you prefer 🙂

Reply:
Yet another alternative:
```
defp report?(%Config{report_transitively_used: true}, _graph, _mfa), do: true
defp report?(_config, graph, mfa), do: Graph.in_degree(graph, mfa) == 0
```

Reply:
rewritten the code to be more 'elixirish' and moved the check on transitively unused calls up to avoid unnecessary membership checks for non root calls